### ISSUE TITLE
Console not writing result to OUTPUT for formats like .ttl, .nt., etc.

### ISSUE THREAD

**Issue by nguyenm100** [eclipse-rdf4j/rdf4j#4893]
### Current Behavior

sparql output="xxx.ttl" <somequery> will complain with a "No suitable result writer found" exception

### Expected Behavior

be able to write sparql results to a turtle file

### Steps To Reproduce

1. windows wsl2
2. run console.sh
3. open up a repo
4. sparql output="abc.ttl" select * where {?s ?p ?o}
5. receive "No suitable result writer found"

### Version

4.3.9 (sdk)

### Are you interested in contributing a solution yourself?

None

### Anything else?

appears only xml, csv, tsv, binary, and maybe another obscure format is supported (anything derived from TupleQueryResultWriterFactory)

**Comment by nguyenm100** [comment]
The side effect is that you can't export inferred triples from console in turtle format

**Comment by hmottestad** [comment]
Try adding the various rio dependencies to the following pom file:

https://github.com/eclipse-rdf4j/rdf4j/blob/main/tools/console/pom.xml

Here is the dependency for turtle: https://mvnrepository.com/artifact/org.eclipse.rdf4j/rdf4j-rio-turtle

If that works maybe you could make a PR that adds all the available rio dependencies to the pom file of the console module?

**Comment by nguyenm100** [comment]
Actually, @hmottestad, there may be some confusion on my part on how this should work.  looks like saving the sparql query results will save as query-results and not as a well formatted turtle (e.g. the format that a sparql endpt returns back to the caller).  This is probably why it only supports xml, csv, json, etc.

However, the actual problem I'm encountering is that I can't seem to export inferred triples (in workbench or console) as it seems inferencing happens only on query and is not persisted.  

Is it possible to do so?

so basically, I create a native+rdfs repo
add some triples
query with inferencing and inferred triples are returned
however if i "export" the repo, inferred triples are not included.

Let me know if I should open another issue.

**Comment by hmottestad** [comment]
Have you tried a CONSTRUCT query?

**Comment by nguyenm100** [comment]
hi Håvard, how does using CONSTRUCT help me serialize an inferred query to file from either console or workbench?

**Comment by hmottestad** [comment]
CONSTRUCT produces triples/quads while a SELECT query produces tuples.

**Comment by nguyenm100** [comment]
I think my question is more fundamental meaning how do i redirect the results of CONSTRUCT to a file (e.g. export) from either console or workbench? realizing i can do all this in code.

**Comment by hmottestad** [comment]
What happens when you run `sparql output="abc.ttl" construct {?s ?p ?o} where {?s ?p ?o}` ?

**Comment by nguyenm100** [comment]
Hey this works, thanks!

I don't believe there's any work to be done here, just clarification on functionality.  Closing. Thanks
